File: Server/src/DatabaseLayout/Models/Voyage.cs
using System;

namespace DatabaseLayout.Models;

public class Voyage
{
    public int Id { get; set; }
    public DateTime VoyageDate { get; set; }

    public int ShipId { get; set; }
    public Ship Ship { get; set; } = null!;

    public int DeparturePortId { get; set; }
    public Port DeparturePort { get; set; } = null!;

    public int ArrivalPortId { get; set; }
    public Port ArrivalPort { get; set; } = null!;

    public DateTime VoyageStart { get; set; }
    public DateTime VoyageEnd { get; set; }
}

File: Client/src/app/models/voyage.ts
export interface Ship {
  id: number;
  name: string;
  maximumSpeed: number;
}

export interface Port {
  id: number;
  name: string;
  country: string;
}

export interface Voyage {
  id: number;
  voyageDate: string;
  voyageStart: string;
  voyageEnd: string;
  ship: Ship;
  departurePort: Port;
  arrivalPort: Port;
}
